circuit/environment/src/helpers/assignment.rs

```diff
@@ -225,6 +227,72 @@ impl<F: PrimeField> snarkvm_algorithms::r1cs::ConstraintSynthesizer<F> for Assig
     }
 }
 
+/// Asserts whether the underlying constraints of the two assignments are equal
```

Comment:
```suggestion
/// Asserts whether the underlying constraints of the two assignments are equivalent in structure
```
Is this a better description?

Reply:
Not quite: the function really checks equality in a pretty dumb way (which is enough for our purposes). For instance, the two linear combinations
`[(Variable(1), 5)]` (i. e. `5 * x_1`)
and
`[(Variable(1), 3), (Variable(1), 2)]` (i. e. `3 * x_1 + 2 * x_1`)
are equivalent, but the function will regard them as different. However, if two different assigments are used to produce the same R1CS, the resulting linear combinations will be equal, i. e. of the same shape (which is stronger than being equivalent), and that is what we check. Checking for equivalence would involve some amount of compilation, which could be very expensive depending on what one checks: take into account that e. g. two R1CS with the same constraints ordered differently are equivalent; as are two R1CS where witness variables are permuted with respect to one another.

Reply:
@d0cd Perhaps "Asserts whether _the representations of_ the constraints underlying the two assignments are equal" would be more fitting?

Reply:
That sounds good!